pandas-stubs/core/frame.pyi

```diff
@@ -487,7 +487,7 @@ class DataFrame(NDFrame, OpsMixin):
     def T(self) -> DataFrame: ...
     def __getattr__(self, name: str) -> Series: ...
     @overload
-    def __getitem__(self, idx: Scalar) -> Series: ...
+    def __getitem__(self, idx: Hashable) -> Series: ...
```

Comment:
can't use `Hashable` here.  It overlaps with `tuple` in the 3rd overload.  Can you put this back to `Scalar` ?

Reply:
Ok, so what is the conclusion here? That we must use `Scalar` to represent the element of an index? Because if that is the case, then I would prefer reverting back all of my `Hashable` changes and only use `Scalar`. I think it would be a bad idea to leave this as `Scalar` and have `__iter__` produce a `Hashable` - that would be an inconsistent annotation and likely cause issues.

IMO - `Hashable` would be more correct. So I would suggest instead that I remove `tuple` from the 3rd overload. In fact, the 3rd overload appears to potentially be inaccurate:

```
import pandas as pd
x = pd.DataFrame([[1, 2, 3]], columns=[(0, 0), (1, 1), (2,2)])
type(x[(0, 0)])
```

> pandas.core.series.Series

It produces a `Series`, not a `DataFrame`, in this context. I can't say whether that is the case in every context.

Reply:
But I think there are additional overlaps with `Hashable`  in the 3rd overload.  Maybe it is the `numpy` objects.  You'd need to do something like `Scalar | tuple` and maybe anything else in that union that is `Hashable`, but not in the 3rd overload.

Reply:
MyPy failures:

> pandas-stubs/core/frame.pyi:495: error: Overloaded function signatures 1 and 3 overlap with incompatible return types  [misc]
tests/test_frame.py:1347: error: Expression is of type "Series[Any]", not "DataFrame"  [assert-type]

I need to do some more investigating. It isn't obvious to me which of the third overload parameters are `Hashable`...

Pyright does NOT report overlapping overloads - which makes me wonder whether it truly is the case or if it's a bug of some kind in MyPy. That said, it does report the second issue:

>   pandas-stubs/tests/test_frame.py:1359:23 - error: "assert_type" mismatch: expected "DataFrame" but received "Series[Unknown]" (reportGeneralTypeIssues)

Will take another look soon